src/battle/anim/water.c

```diff
@@ -882,44 +673,47 @@ _080D3D32:\n\
 }
 #endif
 
-#ifdef NONMATCHING
 void sub_80D3D68(u8 taskId)
 {
     s16 i;
     struct ScanlineEffectParams params;
     struct Task *task = &gTasks[taskId];
-    // u16 *scanlineBuffer;
 
     switch (task->data[0])
     {
         case 0:
             for (i = 0; i < task->data[4]; i++)
             {
-                /* scanlineBuffer = &gScanlineEffectRegBuffers[0][i];
-                *(u16 *)(&gScanlineEffect) = task->data[2];
-                *scanlineBuffer = task->data[2] & -1; */
+                /*
+                   It would be odd for the scanline buffers to follow ASM literal
+                   and be initialized in reverse. Experimentation based upon how
+                   compilers in general load variables and pointers showed that
+                   they were initialized this way.
+
+                   foo = bar = baz -> foo = (bar = baz)
+                */
+                gScanlineEffectRegBuffers[0][i] =
                 gScanlineEffectRegBuffers[1][i] = task->data[2];
```

Comment:
I think breaking these assignments like this is more confusing to read than assigning them in a single line. They're not that long